src/main/java/com/google/jenkins/plugins/k8sengine/KubeConfig.java

```diff
@@ -114,23 +120,25 @@ public Builder currentContext(String currentContext) {
       return this;
     }
 
-    public Builder users(ImmutableList<Object> users) {
+    public Builder users(List<Object> users) {
       config.setUsers(users);
       return this;
     }
 
-    public Builder contexts(ImmutableList<Object> contexts) {
+    public Builder contexts(List<Object> contexts) {
```

Comment:
this does not appear to be binary compatible and is the public interface.  Have you checked for external callers of this function?

Reply:
agree it's a bit agressive I will revert that (even if this really hurt my eyes :))